### DISCUSSION TITLE
Await dont work in BackgroundTaks (Fastapi, Beanie, MongoDB)

### DISCUSSION THREAD

**Discussion by leaguidi** [fastapi/fastapi#8387] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from beanie import PydanticObjectId, Document
from fastapi import FastAPI, APIRouter, Request, BackgroundTasks
from time import time
from pydantic import BaseModel
from typing import Optional

app = FastAPI()
#router = APIRouter(prefix="/audit",tags=['Audit'])

class Audit(Document):
    account: Optional[str]
    company: Optional[str]
    user: Optional[str]
    module: str
    endpoint: str
    url_params: Optional[str]
    body_params: Optional[dict]
    reference_id: Optional[str]
    previous_state: Optional[bytes]
    posterior_state: Optional[bytes]
    date: int = time()

    class Settings:
        name = "audit"

async def review_audits() -> List[Audit]:
       audits = await Audit.find().limit(4).sort([("date", -1),]).to_list()
       print(audits)

@app.get("/query/", response_description="get audits to the database")
async def get_filter_audits(request: Request, background_tasks: BackgroundTasks) -> List[Audit]:
        desc = "We are working on it, you will be notified"
        reviews = {"hash":"12345678"}
        
        background_tasks.add_task(review_audits)
        
        return reviews
```

### Description

when you call de endpoint return the result {"hash":"12345678"} but the background task dont print the result founded on mongodb.
I realized that the error occurs with version 0.85.0 but it works fine with version 0.78.0 

### Operating System

macOS

### FastAPI Version

0.85.0

### Python Version

3.10.9

**Comment by iudeen** [comment]
You are not calling any function inside background Task.

```

@app.post("/send-notification/{email}")
async def send_notification(email: str, background_tasks: BackgroundTasks):
    background_tasks.add_task(write_notification, email, message="some notification")
    return {"message": "Notification sent in the background"}
```

**Comment by JarroVGIT** [comment]
`background_tasks.add_task()` doesn’t add any task.

**Comment by leaguidi** [comment]
Sorry i forgot, insiste o calling the function upside filter_audits

El El vie, 16 de sep. de 2022 a la(s) 14:39, Jarro van Ginkel <
***@***.***> escribió:

> background_tasks.add_task() doesn’t add any task.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/tiangolo/fastapi/issues/5403#issuecomment-1249621613>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHFJEFNS5FQTBWSZ2JGOQH3V6SWGDANCNFSM6AAAAAAQOQK35E>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
-- 

Lea Guidi.

**Comment by leaguidi** [comment]
the code is correct now, sorry for my English.

**Comment by JarroVGIT** [comment]
I cannot reproduce this using the following example:
```python
import asyncio
from fastapi import BackgroundTasks, FastAPI

app = FastAPI()


async def get_items_from_db():
    print("Get some items from DB..")
    await asyncio.sleep(1)
    return [1, 2, 3, 4]


async def background_task_function():
    print("Background task is executing..")
    result = await get_items_from_db()
    print(f"Items collected: {result}")


@app.get("/")
async def root(bt: BackgroundTasks):
    print("Endpoint is called, now adding BG task..")
    bt.add_task(background_task_function)
    print("BG task added, returning response..")
    return {"hello": "world"}


if __name__ == "__main__":
    import uvicorn

    uvicorn.run(app, host="0.0.0.0", port=8000)
```
The above example will print correctly when calling `localhost:8000`:
```
INFO:     Started server process [14322]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)
Endpoint is called, now adding BG task..
BG task added, returning response..
INFO:     127.0.0.1:55946 - "GET / HTTP/1.1" 200 OK
Background task is executing..
Get some items from DB..
Items collected: [1, 2, 3, 4]
```

The issue might be the actual call to `Audit`. Try the following:
- Print something before you call `Audit.find()`, to assert that the background task is actually executing.
- Try to ensure that the `Audit.find()` actually returns something. 

But to answer your issue title: you can certainly use `await` in a background task, and it should work. My money would be on the `Audit.find()` doesn't return anything.

**Comment by leaguidi** [comment]
could be possible, really i don't try to use another thing with await, the
problem are in that line (Audit.find()) but, when i use another version of
fastapi (0.78.0) works fine, the problem is finding in a mongodb database
with beanie and fastapi version 0.85.0.

**Comment by leaguidi** [comment]
I am using versión 0.85.0 no 0.58, i write that tris error no occurs no
versión 0.78.0

El El sáb, 17 de sep. de 2022 a la(s) 18:49, Irfanuddin Shafi Ahmed <
***@***.***> escribió:

> 0.58 is very old. Please update FastAPI :/
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/tiangolo/fastapi/issues/5403#issuecomment-1250145577>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHFJEFPFHKQDMSZLN2XN5FLV6Y4IFANCNFSM6AAAAAAQOQK35E>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>
-- 

Lea Guidi.

**Comment by Kludex** [comment]
> the code is correct now, sorry for my English.

There are people on our discord server that can help you in Spanish: https://fastapi.tiangolo.com/help-fastapi/?h=discord#join-the-chat.

**Comment by vanishingESCkey** [comment]
@leaguidi if you use BaseHTTPMiddleware in your project then it may affect your BackgroundTasks, take a look at issues mentioned in https://github.com/encode/starlette/pull/1715#issue-1289392289

**Comment by thomasleveil** [comment]
@leaguidi @vanishingESCkey a fix for https://github.com/encode/starlette/pull/1715#issue-1289392289 has been released in https://github.com/encode/starlette/releases/tag/0.21.0

Watch  PR #5471
